frontend/src/app/main/ui/ds/penpot/loader.cljs

```diff
@@ -0,0 +1,38 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.ui.ds.penpot.loader
+  (:require-macros
+   [app.common.data.macros :as dm]
+   [app.main.style :as stl])
+  (:require
+   [rumext.v2 :as mf]))
+
+(mf/defc loader*
+  {::mf/props :obj}
+  [{:keys [class width height title] :rest props}]
+  (let [class (dm/str (or class "") " " (stl/css :loader))
+        both-provided (and width height)
+        neither-provided (and (nil? width) (nil? height))
+        props (mf/spread-props props {:viewBox "0 0 677.34762 182.15429"
+                                      :role "status"
+                                      :width (or width "100px")
+                                      :height (or height "27px")
+                                      :class class})]
+    (assert (or both-provided neither-provided)
```

Comment:
asserts need to happen before (there's also no reason we cannot have two lets in a function)